app/services/channel_queues/next_in_queue.rb

```diff
@@ -0,0 +1,32 @@
+module ChannelQueues
+  class NextInQueue
+    attr_accessor :channel_queue
+
+    def initialize(channel_id: nil, channel_name: nil, channel_queue: nil)
+      return unless channel_id || channel_name || channel_queue
+
+      @channel_queue = channel_queue || find_channel_queue(channel_id, channel_name)
+    end
+
+    def user_name
+      user&.slack_user_name
+    end
+
+    def user_id
+      user&.slack_user_id
+    end
+
+    private
+
+    def user
+      channel_queue&.channel_queue_memberships&.order(created_at: :asc)&.joins(:user)&.first&.user
```

Comment:
Seems a little weird to have so many &s since the only call that could fail is the first call on channel_queue. What about a guard clause instead?

Reply:
sure i can ddo that

Reply:
done https://github.com/choruk/foos_portal/compare/7da43fc1b236629a16522d912c39871b2969df88..5ba1b15bef3e5d620e34114a1fd2f9490b9d0f85